apps/admin/backend/src/app.test.ts

```diff
@@ -146,6 +150,10 @@ test('cast vote records - happy path election flow', async () => {
     Admin.CvrFileMode.Unlocked
   );
 
+  // frontend can poll for available cast vote record files
+  mockOf(listCastVoteRecordFilesOnUsb).mockResolvedValueOnce([]);
+  expect(await apiClient.listCastVoteRecordFilesOnUsb()).toEqual([]);
```

Comment:
This approach to testing is what I would call a "tower of mocks." Each layer in the call stack gets unit tested with the layer immediately below mocked. I personally try to avoid these kinds of towers because I think they are brittle to changes. If any code in the stack gets refactored, many of the tests and mocks in the tower may have to change. In particular, it can be dangerous if I change a function's behavior and then forget to update a mock of that function in another test somewhere.

Instead, I try to stretch out my tests as much as possible so they are testing from the highest interface in the call stack that makes sense (e.g. in this case, the API) and mock the lowest interface in the call stack as possible (in this case, the USB drive itself). That gives me the most freedom to refactor the code in between with the tests as my safeguard.

I don't know that everybody agrees with me on this take, and it's certainly got some tradeoffs, but figured I'd share it and see what you thought.

Reply:
I don't prefer this style of testing either. When I worked on the VxAdmin `grout` transition, I tried to remove any dependence on `store` mocking and just test with the whole `store` running.

I mostly mocked in this way because it was easier, and the lower layers are a little harder to mock. Since `getUsbDrives` is internal to `libs/backend`, I can't mock it on its own. I have to mock the touchpoint with that module, `listDirectoryOnUsbDrive`. I could mock that here in `app.test.ts` as well, but for whatever reason I also deemed that to be more work.

To really get rid of the tower here we'd have to change `listDirectory` to accept `getUsbDrives` as an argument, similar to how the `Exporter` that @eventualbuddha created works, and mock that. That would be the more "testable" code pattern.

I agree with your calling this out but on its own it doesn't seem quite important enough to block and spend time on, but I don't feel strongly and can mock this differently if you do think it's important.

Reply:
Maybe there's a way we can set up the mocking for `getUsbDrives` once and reuse it? I already made an attempt at this here: https://github.com/votingworks/vxsuite/blob/1f3cbf7c4d8c480f768c2ac1f06065442316425a/apps/scan/backend/test/helpers/app_helpers.ts#L73. In that case I used dependency injection like you mentioned.

Talking this through, I think a good path forward is to have as thin as possible of an abstraction layer over the actual filesystem commands that are talking to the USB drive and then mock that.

I don't feel strongly about whether you move towards that in this PR.

Reply:
I understand this `createMockUsb` and the dependency injection - that seems like a good path forward here. But I'm not sure what you mean by the following: 

> Talking this through, I think a good path forward is to have as thin as possible of an abstraction layer over the actual filesystem commands that are talking to the USB drive and then mock that.

What do you mean by that? Something different than `createMockUsb` is doing?

Reply:
I think by that last part I just was reiterating the same thing that `Usb` and `createMockUsb` is already doing. Sorry for being confusing lol